src/compiler/transformers/declarations.ts

```diff
@@ -1927,17 +1929,20 @@ export function transformDeclarations(context: TransformationContext) {
                     input.name,
                     factory.createNodeArray(mapDefined(input.members, m => {
                         if (shouldStripInternal(m)) return;
+                        // Rewrite enum values to their constants, if available
+                        const constValue = resolver.getConstantValue(m);
                         if (isolatedDeclarations) {
                             if (
-                                m.initializer && !resolver.isLiteralConstDeclaration(m) &&
+                                m.initializer &&
                                 // This will be its own compiler error instead, so don't report.
-                                !isComputedPropertyName(m.name)
+                                !isComputedPropertyName(m.name) &&
+                                (constValue === undefined ||
+                                    ((isIdentifier(m.initializer) || isPropertyAccessExpression(m.initializer)) &&
+                                        !isExprEnumMember(input, m.initializer)))

```

Comment:
Are we sure this covers all allowed cases?  An enum `enum E { A = 1, B = A << 1 }` should be allowed. And it doesn't seem from this code like it is.

Reply:
But what if the value is statically resolvable, comes from somewhere else, but is also an expression with operators. 

```ts
enum E { A = 1, B = OtherEnum.B << 1, C = A << 1 }
```

Both `B` and `C` are statically resolvable. But `C` is allowed as it references only the enum itself. `B` is not  because it references `OtherEnum`. I don't understand how TSC will error on one but not the other.  I don't see how the code there walks the expression, and allows the `<<` in the first case, but not the second. Maybe I'm missing something.

Reply:
You're totally right, for this example it doesn't work as it's intended to be. 
 So in tsc with --ID and --declaration, this example will be emittted without an error, of which it should rather complain on `B`. In our custom emitter, this errors on `B` as inteded. 

> I don't understand how TSC will error on one but not the other. 
I didn't understand what you're confused about, but if you're asking where the discrepancy comes from, it's coming from the difference in the emitResolver. DTE is calling in [getConstantValue](https://github.com/bloomberg/TypeScript/blob/isolated-declarations/src/compiler/transformers/declarations/emit-resolver.ts#L327), which underlying is calling the [evaluate](https://github.com/bloomberg/TypeScript/blob/isolated-declarations/src/compiler/transformers/declarations/evaluator.ts#L21) which discriminates expressions that refers to symbols within current enum scope and others. The call path of the regular TSC is calling the regular resolver from the checker.ts:evaluate ( too big to have a link on the file) which doesn't discrimiate those two.

Either way, the current behavior of this PR is wrong as you said. I think I have to align two evaluators after all as you said.

Reply:
Updated the evaluator.